### PR TITLE
[A-3]courses_management-abdulmjid

### PR DESCRIPTION
**Pull request by XMjeedX**
program that allows students to (add courses, drop courses, list courses).

### CODE REVIEW

File: courses_management/application.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def run(self):
+        while True:
+            print("\nMain Menu")
+            print("1. Add Course")
+            print("2. Drop Course")
+            print("3. List Courses")
+            print("4. Exit")
+            choice = input("Enter your choice (1-4): ")
+
+            if choice == '1':
```
Comment (MohammadModallalBloovo) [inline courses_management/application.py:53]: Enum

File: courses_management/config.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[slack]
```
Comment (MohammadModallalBloovo) [inline courses_management/config.ini:1]: remove

File: courses_management/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+asttokens==2.4.1
+colorama==0.4.6
```
Comment (MohammadModallalBloovo) [inline courses_management/requirements.txt:2]: check

File: courses_management/application.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+from course_registry import CourseRegistry
+from csv_file_mange import csv_file_mange
+from utils.slack import send_slack_notification
+from utils.constants import MENU_CHOICES
```
Comment (MohammadModallalBloovo) [inline courses_management/application.py:4]: add spaces, check python files structure

File: courses_management/application.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    def run(self):
+        while True:
+            print("\nMain Menu")
+            print("1. Add Course")
+            print("2. Drop Course")
+            print("3. List Courses")
+            print("4. Exit")
+            choice = input("Enter your choice (1-4): ")
+
+            if choice == MENU_CHOICES['Add_Course']:
```
Comment (MohammadModallalBloovo) [inline courses_management/application.py:53]: implement python Enum

File: courses_management/csv_file_mange.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import csv
+import utils.constants as constants
+
+class csv_file_mange:
+    @staticmethod
+    def save_courses_to_csv(courses):
```
Comment (MohammadModallalBloovo) [inline courses_management/csv_file_mange.py:6]: not ready

File: courses_management/utils/slack.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import json
+import configparser
+from .constants import CONFIG_PATH
+
+def load_config():
+    config = configparser.ConfigParser()
+    config.read(CONFIG_PATH)
+    return config['slack']['webhook_url']
+
+def send_slack_notification(message):  # Default color set to green
```
Comment (MohammadModallalBloovo) [inline courses_management/utils/slack.py:11]: reimplement Slack

**Review by MohammadModallalBloovo — APPROVED** [review]

File: courses_management/utils/constants.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+from enum import Enum, auto
+from datetime import datetime
+
+COURSES_CSV_PATH = 'courses.csv'
```
Comment (MohammadModallalBloovo) [inline courses_management/utils/constants.py:4]: Add it to "data".

File: courses_management/csv_file_mange.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import csv
```
Comment (MohammadModallalBloovo) [inline courses_management/csv_file_mange.py:1]: repository/course_repository.py

File: courses_management/utils/slack.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import requests
+import json
+from datetime import datetime
+
+class SlackNotifier:
+    def __init__(self):
+        # Fetch the webhook URL from the environment variable
+        self.webhook_url = os.getenv('SLACK_WEBHOOK_URL')
+    
+    def send_notification(self, message, color="#FF0000", pretext="Notification", author_name="Course Mangmet System"):
```
Comment (MohammadModallalBloovo) [inline courses_management/utils/slack.py:11]: no need for the default values, only keep "color"e

**Review by ZDZD-S — COMMENTED** [review]
Everything is okay

**Review by ZDZD-S — COMMENTED** [review]
Everything is okay

**Review by ZDZD-S — APPROVED** [review]

File: courses_management/utils/constants.py [hunk 11]
```diff
@@ -0,0 +1,5 @@
+from datetime import datetime
+
+COURSES_CSV_NAME = 'courses_management/data/courses.csv'
```
Comment (ZDZD-S) [inline courses_management/utils/constants.py:3]: ??

File: courses_management/repository/course_repository.py [hunk 12]
```diff
@@ -0,0 +1,17 @@
+from utils.errors import COURSE_ALREADY_EXISTS_ERROR, COURSE_NOT_FOUND_ERROR
```
Comment (FAISAL-K-ALOTAIBI) [inline courses_management/repository/course_repository.py:1]: you need to have space between the module and the class.

**Review by FAISAL-K-ALOTAIBI — APPROVED** [review]

**Review by ZDZD-S — APPROVED** [review]

File: courses_management/application.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
+from repository.course_repository import CourseRegistry
+from utils.csv_file_helper import CSVFileManager
+from utils.notification import SlackNotifier 
+from enums.menu_choices import MenuChoices
+
+class Application:
+    # Initializes the application and loads courses.
```
Comment (FAISAL-K-ALOTAIBI) [inline courses_management/application.py:7]: best p for comments

File: courses_management/utils/notification.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
+import os
+import requests
+import json
+
+class Notifier:
```
Comment (FAISAL-K-ALOTAIBI) [inline courses_management/utils/notification.py:5]: ??

**Review by FAISAL-K-ALOTAIBI — APPROVED** [review]

**Review by ZDZD-S — APPROVED** [review]

**Pull request merged**